File: 2000 98 & ME/QuickStart/howto/samples/Services/PerformanceCounters/pcdemo/pcdemo.cs
using System;
using System.Diagnostics;
using System.Timers;
using System.Threading;

public class PCDemo{

	static PerformanceCounter theCounter;

	public static void Main(string[] args){

		string objectName   = "ACounterDemo";
		string counterName  = "CountPerSecond";
		string instanceName = "_Total";

		if(!PerformanceCounter.CategoryExists(objectName))
		{
			CounterCreationData ccd = new CounterCreationData();
			ccd.CounterName = counterName;
			ccd.CounterType = PerformanceCounterType.RateOfChangePerSecond32;
			CounterCreationData[] ccds = new CounterCreationData[1];
			ccds[0] = ccd;

			PerformanceCounterCategory.Create(objectName,"Sample Object",ccds);
		}
		
		theCounter = new PerformanceCounter(objectName, counterName ,instanceName);

		System.Timers.Timer aTimer = new System.Timers.Timer();
		aTimer.Tick += new EventHandler(OnTimer);

		aTimer.Interval  = 100;
		aTimer.Enabled   = true;
		aTimer.AutoReset = false;
		
		Console.WriteLine("Press \'+\' to increase the interval");
		Console.WriteLine("Press \'-\' to decrease the interval");
		Console.WriteLine("Press \'q\' to quit the sample");
		Console.WriteLine("Started");

		int command;
		do
		{
			command = Console.Read();
			if(command=='+')
				aTimer.Interval = Math.Max(1,aTimer.Interval/2);
			if(command=='-')
				aTimer.Interval *= 2;


			Thread.Sleep(500);
		}
		while(command!='q');
		PerformanceCounter.DeleteCategory(objectName);
	}

	public static void OnTimer(Object source, EventArgs e)
	{
		theCounter.IncrementBy(1);
		System.Timers.Timer theTimer = (System.Timers.Timer)source;
		theTimer.Enabled = true;
	}
}

File: 2000 98 & ME/QuickStart/howto/samples/Services/PerformanceCounters/pcdemo/vb/pcdemo.vb
Imports System
Imports System.Diagnostics
Imports System.Timers
Imports System.Threading

Public class PCDemo

	shared theCounter as PerformanceCounter

	public shared sub Main()
		Dim objectName As String = "ACounterDemo"
		Dim counterName As String = "CountPerSecond"
		Dim instanceName As String = "_Total"
		
		if(not PerformanceCounter.CategoryExists(objectName)) then
			Dim ccd As CounterCreationData = new CounterCreationData()
			ccd.CounterName = counterName
			ccd.CounterType = PerformanceCounterType.RateOfChangePerSecond32
			Dim ccds(1) As CounterCreationData
			ccds(0) = ccd

			PerformanceCounterCategory.Create(objectName,"Sample Object",ccds)
		end if
		
		theCounter = new PerformanceCounter(objectName, counterName ,instanceName)

		Dim aTimer As new System.Timers.Timer
		AddHandler aTimer.Tick, AddressOf OnTimer

		aTimer.Interval = 100
		aTimer.Enabled = true
		aTimer.AutoReset = false

		Console.WriteLine("Press '+' to increase the interval")
		Console.WriteLine("Press '-' to decrease the interval")
		Console.WriteLine("Press 'q' to quit the sample")
		Console.WriteLine("Started")
		
		Dim command As Integer
		do
			command = Console.Read()
			if(command=43) then			' + = ASCII 43
				aTimer.Interval = Math.Max(1,aTimer.Interval/2)
			elseif(command=45) then		' - = ASCII 45
				aTimer.Interval *= 2
			end if

		loop until command=113
		
		PerformanceCounter.DeleteCategory(objectName)	  
	end sub
	
	public shared sub OnTimer(source as Object, e As EventArgs)
		theCounter.IncrementBy(1)
		Dim theTimer As System.Timers.Timer = CType(source,System.Timers.Timer)
		theTimer.Enabled = true
	end sub
end class